### ISSUE TITLE
Fix request: Avoid identity cache keys false positive: other params usage

### ISSUE THREAD

**Issue by DungBe** [jborgers/PMD-jPinpoint-rules#133]
Avoid identity cache keys by casting the generate method parameters,  Check also when params are used for logging, the rule gives a violation, too, which is a false positive.
```
  @NotNull
    @Override
    public Object generate(Object target, Method method, Object... params) {
        if (params.length < 2 ) {
            String errorMessage = MessageBuilder
                    .format("KeyGenerator assumes at least 2 parameter: {}",
                            params);
            throw new IllegalArgumentException(errorMessage);
        }
        String relationId = (String) params[0];
        String securitiesAccountId = (String) params[1];
        return new SimpleKey(appName, ClassUtils.getUserClass(target.getClass()).getName(), relationId, securitiesAccountId);
    }
```

**Comment by jborgers** [comment]
Added: not a violation when params is used within an if-statement checking for length.
